zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/FlownodeInstanceQueryController.java

```diff
@@ -51,11 +52,18 @@ private ResponseEntity<FlownodeInstanceSearchQueryResponse> search(
               .search(query);
       return ResponseEntity.ok(
           SearchQueryResponseMapper.toFlownodeInstanceSearchQueryResponse(result));
-    } catch (final Throwable e) {
+    } catch (final ValidationException e) {
       final var problemDetail =
           RestErrorMapper.createProblemDetail(
               HttpStatus.BAD_REQUEST,
               e.getMessage(),
+              "Validation failed for Flownode Instance Search Query");
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    } catch (final Throwable e) {
```

Comment:
I suggest we log this unexpected exception stack trace. This can be helpful in troubleshooting, and `e.getMessage `may not be explicit enough
```
REST_LOGGER.warn("An exception occurred in searchFlowNodeInstances.", e);
```